### PR TITLE
First Push of HMOS ported library

### PR DESCRIPTION
**Pull request by Click2cloud-Eros**
Add HMOS ported library.

### CODE REVIEW

File: xradar/src/main/java/com/orzangleli/radar/TextPaint.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package com.orzangleli.radar;
+
+import ohos.agp.render.Paint;
+
+public class TextPaint extends Paint {
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/TextPaint.java]: Remove unused declarations in this and other classes.

File: xradar/src/main/java/com/orzangleli/radar/RectF.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+package com.orzangleli.radar;
+
+import ohos.agp.utils.RectFloat;
+import ohos.media.image.common.Rect;
+
+/**
+ * RectF class
+ */
+public class RectF extends RectFloat {
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/RectF.java]: So many unused declarations

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java [hunk 3]
```diff
@@ -541,38 +616,47 @@ private void drawRegionWithColor(Canvas canvas, float scale) {
...
+     * @param drawable      icon drawable
+     * @param verticalValue vertical value
+     * @param position      position
+     */
+    private void drawMultiLinesTextAndIcon(Canvas canvas, float x, float y, CharSequence text, int drawable, int verticalValue, int position) {
         int drawableAvaiable = 1;
         try {
-            this.getContext().getResources().openRawResource(drawable);
+            this.getContext().getResourceManager().getResource(drawable);
             drawableAvaiable = 1;
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/XRadarView.java:770]: No use of this

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java [hunk 4]
```diff
@@ -374,9 +423,14 @@ private void drawPolygon(Canvas canvas) {
...
-    private void drawLayer(Canvas canvas, int startColor, int endColor) {
+    /**
+     * Draw the colors of each layer
+     *
+     * @param canvas     instance of Canvas
+     * @param startColor start color
+     * @param endColor   end color
+     */
+    private void drawLayer(Canvas canvas, Color startColor, Color endColor) {
         Path path = null;
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Null initialization is redundant here, there are similar issues at other places too, please fix them all.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java [hunk 5]
```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
...
+    /**
+     * RefreshListener updates when view is relayout
+     */
+    class RefreshListener implements LayoutRefreshedListener {
+        @Override
+        public void onRefreshed(Component component) {
+            int w = component.getWidth();
+            int h = component.getHeight();
+            radius = Math.min(h, w) / 2 * radarPercent;
+            MAX_TEXT_WIDTH = (int) (Math.min(h, w) / 2 * (1 - radarPercent));
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Here there is a numeric issue - integer division in the floating-point context. Also, similar issues on several places, pls fix all.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java [hunk 6]
```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
...
-    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
-        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
-        int wSpecSize = MeasureSpec.getSize(widthMeasureSpec);
-        int hSpecSize = MeasureSpec.getSize(heightMeasureSpec);
-
-        setMeasuredDimension(Math.min(wSpecSize, hSpecSize), Math.min(wSpecSize, hSpecSize));
-    }
+    public boolean onTouchEvent(Component component, TouchEvent touchEvent) {
+        HiLog.info(LABEL_LOG, "%{public}s", "onTouchEvent called");
+        switch (touchEvent.getAction()) {
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: 'switch' statement has too few case labels (1), and should probably be replaced with an 'if' statement

File: entry/src/main/resources/base/layout/ability_main.xml [hunk 7]
```diff
@@ -0,0 +1,478 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (kanaksony) [inline entry/src/main/resources/base/layout/ability_main.xml:1]: So many Hardcoded strings in this XML, should use `$string: ` resource

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java [hunk 8]
```diff
@@ -1,126 +1,136 @@
...
-import android.view.View;
-import android.view.animation.LinearInterpolator;
+import ohos.agp.animation.Animator;
+import ohos.agp.animation.AnimatorValue;
+import ohos.agp.colors.RgbPalette;
+import ohos.agp.components.AttrSet;
+import ohos.agp.components.Component;
+import ohos.agp.render.*;
+import ohos.agp.text.SimpleTextLayout;
+import ohos.agp.utils.*;
```
Comment (kanaksony) [inline xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Instead, we can import explicit classes

**Review by kanaksony — CHANGES_REQUESTED** [review]
So many places - proper indentation, extra and required spacing, code reformating required.

File: xradar/src/main/java/com/orzangleli/radar/TextPaint.java (hunk 1 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/TextPaint.java]: In TextPaint class not a single method is used, So can I remove this class and directly use super class(i.e. Paint) ?

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java (hunk 3 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/XRadarView.java:770]: `this.getContext().getResourceManager().getResource(drawable);`

**Use of this line:**
if this line throw exception then, drawableAvailable = 0 else drawableAvailable = 1

**Comment by Click2cloud-Eros** [comment]
All changes Done.

File: xradar/src/main/java/com/orzangleli/radar/TextPaint.java (hunk 1 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/TextPaint.java]: Remove TextPaint class.

File: xradar/src/main/java/com/orzangleli/radar/RectF.java (hunk 2 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/RectF.java]: Remove RectF class.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java (hunk 4 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Remove null redundancy.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java (hunk 5 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: solved numeric warning - integer division in the floating-point context.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java (hunk 6 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Switch statement replaced by if statement.

File: entry/src/main/resources/base/layout/ability_main.xml (hunk 7 above)
Comment (Click2cloud-Eros) [reply entry/src/main/resources/base/layout/ability_main.xml:1]: Move hardcoded string to string.json resource.

File: xradar/src/main/java/com/orzangleli/radar/XRadarView.java (hunk 8 above)
Comment (Click2cloud-Eros) [reply xradar/src/main/java/com/orzangleli/radar/XRadarView.java]: Done.

**Pull request merged**